azurelinuxagent/common/protocol/wire.py

```diff
@@ -1093,10 +1090,11 @@ def send_event(self, provider_id, event_str):
 
     def report_event(self, event_list):
         buf = {}
+        events_per_request = 0
         # Group events by providerId
         for event in event_list.events:
             if event.providerId not in buf:
-                buf[event.providerId] = ""
+                buf[event.providerId] = b''
```

Comment:
Converted the buffer strings to `bytes`. This is mainly needed for py3 as `str.encode()` converts to `bytes`. In Py2, `type(bytes) == type(str)` so this is a no-op.